### PR TITLE
Provide proper error handing when an IDP cannot be retrieved via the WebID

### PR DESCRIPTION
**Pull request by svdwyer**
…f the WebID doesn't exist or the IDP cannot be determined

- show error message when IDP cannot be retrieved
- allow message to be discarded
- added some e2e tests for the IDP retrieval

### CODE REVIEW

**Comment by pheyvaer** [comment]
Can you give this PR a proper name, please?

File: tests/main.spec.js [hunk 1]
```diff
@@ -416,3 +435,62 @@ test('edits of WebID are persisted', async ({ popupPage }) => {
 
   await expect(webid).toHaveValue('NEW WEBID');
 });
+
+test('popup starts with the error section hidden', async ({ page }) => {
+  await expect(page.locator('#error-message-container')).toBeHidden();
+  await expect(page.locator('#error-message-container')).toBeHidden();
```
Comment (pheyvaer) [inline tests/main.spec.js:441]: Why is this happening a second time?

File: src/js/sw.js [hunk 2]
```diff
@@ -111,10 +109,20 @@ const handleInternalMessage = async (message) => {
 
     let idpOrWebID = activeIdentity.idp;
     if (activeIdentity.webID) {
-      const idps = await getIDPsFromWebID(activeIdentity.webID);
-      // In the end we set the IDP anyway. User gets redirected to IDP. If it supports more than one WebID, the user can confirm or select the correct one there.
-      if (idps.length) {
-        idpOrWebID = idps[0];
+      try {
+        const idps = await getIDPsFromWebID(activeIdentity.webID);
+        // In the end we set the IDP anyway. User gets redirected to IDP. If it supports more than one WebID, the user can confirm or select the correct one there.
```
Comment (pheyvaer) [inline src/js/sw.js:114]: I don't understand what this comment is clarifying.

File: README.md [hunk 3]
```diff
@@ -169,6 +169,7 @@ Perhaps with additional research, a workaround can be conceptualized that does n
 To run end-to-end tests, you must first have built the extension (`npm run build`).
 Then the Playwright tests can be run with `npx playwright test` for headless mode.
 If you want to run the tests interactively, you can use `npx playwright test --ui`.
+To mock the network requests from the service worker an environment variable must be set: PW_EXPERIMENTAL_SERVICE_WORKER_NETWORK_EVENTS=1.
```
Comment (pheyvaer) [inline README.md:172]: Can we put this all in a single script in `package.json`?

**Review by pheyvaer — CHANGES_REQUESTED** [review]

File: tests/main.spec.js (hunk 1 above)
Comment (svdwyer) [reply tests/main.spec.js:441]: should only happen once

File: src/js/sw.js (hunk 2 above)
Comment (svdwyer) [reply src/js/sw.js:114]: I'm not sure I understand either. I think the comment should state that if multiple IDPs are available and that by default the first one will be selected.
Agree?

File: README.md (hunk 3 above)
Comment (svdwyer) [reply README.md:172]: yes, that is possible. will do

Comment (pheyvaer) [reply README.md:172]: Can you also update the README to explain to use `npm test`?

Comment (svdwyer) [reply README.md:172]: sure. shall I also add a script to run the tests in UI mode / interactively?

Comment (pheyvaer) [reply README.md:172]: Sure.

File: src/popup/popup.js [hunk 4]
```diff
@@ -31,6 +47,10 @@ const handleInternalMessage = (message) => {
     return;
   }
 
+  if (message.type === 'active-identity-response-error') {
+    handleError('Unable to retrieve IDP from WebID.');
```
Comment (pheyvaer) [inline src/popup/popup.js:51]: 1. Mention the profile so users know what profile the extension is talking about.
2. The error message should disappear when a working profile is selected.

File: README.md (hunk 3 above)
Comment (svdwyer) [reply README.md]: done

File: src/js/sw.js (hunk 2 above)
Comment (pheyvaer) [reply src/js/sw.js:114]: If we neither of us understand it, remove it then.

File: README.md [hunk 5]
```diff
@@ -169,6 +169,9 @@ Perhaps with additional research, a workaround can be conceptualized that does n
 To run end-to-end tests, you must first have built the extension (`npm run build`).
 Then the Playwright tests can be run with `npx playwright test` for headless mode.
 If you want to run the tests interactively, you can use `npx playwright test --ui`.
```
Comment (pheyvaer) [inline README.md:171]: If you add the scripts, then this can be removed.

File: README.md [hunk 6]
```diff
@@ -169,6 +169,9 @@ Perhaps with additional research, a workaround can be conceptualized that does n
...
+To mock the network requests from the service worker an (currently experimental) Playwright environment variable must be set: PW_EXPERIMENTAL_SERVICE_WORKER_NETWORK_EVENTS=1.
```
Comment (pheyvaer) [inline README.md:172]: Did you run the Markdown linter?

**Review by pheyvaer — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (pheyvaer) [reply README.md:171]: This is not fixed yet.

**Review by pheyvaer — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -168,7 +168,9 @@ Perhaps with additional research, a workaround can be conceptualized that does n
 
 To run end-to-end tests, you must first have built the extension (`npm run build`).
 Then the Playwright tests can be run with `npx playwright test` for headless mode.
-If you want to run the tests interactively, you can use `npx playwright test --ui`.
+To mock the network requests from the service worker an (currently experimental) Playwright environment
+variable must be set: PW_EXPERIMENTAL_SERVICE_WORKER_NETWORK_EVENTS=1.
+This is automatically set when running the tests with `npm run test`.
```
Comment (pheyvaer) [inline README.md:173]: I want to see something like this actually:

> To run end-to-end tests, you must first have built the extension (`npm run build`).
Then you run the tests via `npm test`.
If you want to run the tests interactively, you use `npm run test:ui`.

There is no need to add the details of the ENV parameter and what command the scripts execute. If people want to know they can check the `package.json`.

**Review by pheyvaer — CHANGES_REQUESTED** [review]

**Review by pheyvaer — APPROVED** [review]

**Pull request merged**